Bump versions

alexander-yevsyukov opened the pull request:
This PR:
  * Updates `config` dependency.
  * Bumps `base` dependency.
  * Adds GitHub workflow step for uploading Codecov reports.

alexander-yevsyukov committed:
Update config

alexander-yevsyukov committed:
Add step for updating Codecov

alexander-yevsyukov committed:
Bump versions

alexander-yevsyukov commented:
Going to separate Codecov thing into a separate GitHub Action-based check.

alexander-yevsyukov committed:
Turn Codecov patch status on

yuri-sergiichuk commented on buildSrc/build.gradle.kts:
```diff
@@ -55,12 +55,30 @@ repositories {
     mavenLocal()
     gradlePluginPortal()
     mavenCentral()
+    maven {
+        url = uri("https://repo.spring.io/libs-release")
+    }
```
This should also work:
```suggestion
    maven("https://repo.spring.io/libs-release")
```

Also, maybe worth adding a note on why do we need Spring repos here?

yuri-sergiichuk replied:
@alexander-yevsyukov `buildSrc` isn't completely "copied". The `build.gradle.kts` belongs to this project.

alexander-yevsyukov replied:
@armiol, can you shed some light on Spring matter, please? I also had this question, but was lazy enough to get down to exploring it under `config`. Now, after Yuri's comments I'm there... :)

After your answer I'll update `config` accordingly.

alexander-yevsyukov replied:
@armiol, do we need Spring Maven repo for this dependency `org.ajoberstar.grgit.Grgit`?

 They specify another Maven repo in their project page (https://github.com/ajoberstar/grgit).

The problem is that I don't see `ajoberstar` at Spring Maven repo:
https://repo.spring.io/ui/native/libs-release/org

alexander-yevsyukov replied:
https://ajoberstar.org/bintray-backup/ says it's only for older versions of the tool.

yuri-sergiichuk commented on buildSrc/src/main/groovy/license-report-project.gradle:
```diff
@@ -80,8 +81,10 @@ class MarkdownReportRenderer implements ReportRenderer {
         config = project.licenseReport
         output = new File(config.outputDir, fileName)
+        final publishExtension = project.rootProject.extensions.findByType(PublishExtension.class)
+        final prefix = (publishExtension?.spinePrefix?.getOrElse(false) ?: false) ? "spine-" : ""
```
I'm sorry, but this is way too much...

yuri-sergiichuk replied:
Here's a PR to address this: https://github.com/SpineEventEngine/config/pull/242.

alexander-yevsyukov replied:
This is being addressed in https://github.com/SpineEventEngine/config/pull/243.

yuri-sergiichuk reviewed (approved):
@alexander-yevsyukov I'm approving the PR, but I guess we should address the ternary hellish condition noted in the comment.

alexander-yevsyukov commented:
@yuri-sergiichuk, I understand your concerns, but I don't think I can address them just like that in this PR.

`base-types` is downstream to `config` as the rest of repositories. Addressing it here does not make much sense, as it's going to be overwritten by next `config` update.

How do you suggest to address your comments related to files under `buildSrc`?

alexander-yevsyukov committed:
Allow success Codecov threshold

... to avoid false negatives because of rounding issues in Codecov.

alexander-yevsyukov committed:
Set Codecov status to `error` when CI failed

alexander-yevsyukov committed:
Remove Codecov step

alexander-yevsyukov committed:
Add workflow for Codecov

alexander-yevsyukov committed:
Bump `config`

alexander-yevsyukov committed:
Add Codecov step back

alexander-yevsyukov committed:
Remove workflow

alexander-yevsyukov committed:
Set coverage report dir explicitly

alexander-yevsyukov committed:
Try codecov-action@v1

alexander-yevsyukov committed:
Back to `codecov-action@v2`

alexander-yevsyukov committed:
Update config

alexander-yevsyukov committed:
Remove explicit directory setup

alexander-yevsyukov committed:
Temporarily remove the Codecov upload step

It fails because Jacoco reports are not generated during the build. To be addressed in a separate PR.

alexander-yevsyukov committed:
Fix generating Jacoco reports
